app/src/main/java/org/mifos/mobile/ui/registration/RegistrationScreen.kt

```diff
@@ -252,36 +371,214 @@ fun RegistrationScreen(
 
         Button(
             onClick = {
-                register.invoke(
+                val error = register.invoke(
```

Comment:
check for errors directly in here
`val error = areFieldsValidated( pass things)`

Reply:
only invoke register if there is no error